app/src/main/java/io/intrepid/contest/screens/contestcreation/NewContestActivity.java

```diff
@@ -0,0 +1,142 @@
+
+import java.util.List;
+import java.util.UUID;
+
+import butterknife.BindView;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseMvpActivity;
+import io.intrepid.contest.base.PresenterConfiguration;
+import io.intrepid.contest.models.Category;
+import io.intrepid.contest.models.Contest;
+import io.intrepid.contest.screens.contestcreation.addcategoriestocontest.AddCategoriesFragment;
+import io.intrepid.contest.screens.contestcreation.describecontest.DescribeContestFragment;
+import io.intrepid.contest.screens.contestcreation.namecontest.NameContestFragment;
+import io.intrepid.contest.screens.entrysubmission.entryimage.EntryImageActivity;
+import io.intrepid.contest.screens.splash.SplashActivity;
+import io.intrepid.contest.utils.SlidingTabAdapter;
+import timber.log.Timber;
+
+public class NewContestActivity extends BaseMvpActivity<NewContestPresenter> implements NewContestContract.View, ContestActivityContract{
+
+    private static final String KEY_CREATOR_ID = "User_Id";
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.fragment_container)
+    ViewPager viewPager;
+    private SlidingTabAdapter tabAdapter;
+
+    public static Intent createIntent(Context context, UUID creatorId) {
+        return new Intent(context, NewContestActivity.class).putExtra(KEY_CREATOR_ID, creatorId);
+    }
+
+    @NonNull
+    @Override
+    public NewContestPresenter createPresenter(PresenterConfiguration configuration) {
+        return new NewContestPresenter(this, configuration);
+    }
+
+    @Override
+    protected int getLayoutResourceId() {
+        return R.layout.activity_viewpager_container;
+    }
+
+    public NewContestContract.Presenter getPresenter() {
+        return presenter;
+    }
+
+    public void setupViewPager(ViewPager viewPager) {
+        setToolbarTitle(R.string.new_contest);
+        tabAdapter = new SlidingTabAdapter(this);
+        tabAdapter.addFragment(new NameContestFragment());
+        tabAdapter.addFragment(new DescribeContestFragment());
+        tabAdapter.addFragment(new AddCategoriesFragment());
+        viewPager.setAdapter(tabAdapter);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        menu.clear();
+        getMenuInflater().inflate(R.menu.menu_new_contest, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                Timber.d("On back pressed ");
+                presenter.onBackButtonClicked();
+                break;
+            case R.id.action_next:
+                presenter.onNextButtonClicked();
+                break;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    public void setToolbarTitle(@StringRes int toolbarTitle) {
+        setSupportActionBar(toolbar);
+        ActionBar actionBar = getSupportActionBar();
+        if(actionBar != null) {
```

Comment:
where would be the best place to implement this setup kinda stuff? The activity's onCreate() method is final so it can't be overridden, and I don't think this needs to be something that the presenter delegated back to the activity. @jmccormack200 @gsintrepid